### PR TITLE
[REVITALIZATION] Test progress events

### PR DESCRIPTION
**Pull request by kepta**
This pull request aims to add progress events under test coverage.

### CODE REVIEW

**Comment by davidtheclark** [comment]
@kepta Yeah, we might just have to live with those differences — in which case we'll want to implement separate tests for each environment. 

For the browser, I think using XHR's native events is fine, even if they sometimes behave kind of strangely. Unless you can figure out a better way that is also relatively simple?

**Comment by kepta** [comment]
@davidtheclark I have segregated the node and browser testing and I tried using the `xhr-mock` library for the browser tests.

File: lib/classes/mapi-request.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ var parseToken = require('@mapbox/parse-mapbox-token');
 var xtend = require('xtend');
 var EventEmitter = require('eventemitter3');
 var urlUtils = require('../helpers/url-utils');
-
+var constants = require('../constants');
 var requestId = 1;
```
Comment (davidtheclark) [inline lib/classes/mapi-request.js]: I'd prefer to conventionally keep an empty line separating imports from the rest of the code.

File: test/browser-interface.test.js [hunk 2]
```diff
@@ -19,3 +20,59 @@ test('errors early if access token not provided', () => {
     }
   );
 });
+
+describe('test node progress events', () => {
```
Comment (davidtheclark) [inline test/browser-interface.test.js]: node > browser

File: test/browser-interface.test.js [hunk 3]
```diff
@@ -19,3 +20,59 @@ test('errors early if access token not provided', () => {
...
+
+  beforeEach(() => {
+    mockXHR.setup();
+    const accessToken = mockToken();
+    mockXHR.get(
+      `https://api.mapbox.com/styles/v1/mockuser/foo?access_token=${accessToken}`,
+      {
+        status: 200,
+        headers: {
+          'Content-Length': '18',
```
Comment (davidtheclark) [inline test/browser-interface.test.js]: Would you mind adding a comment with the URL of the mockXhr docs that explain why this works? I expect I'd be confused by this if I look at it again 4 months later.

File: test/browser-interface.test.js [hunk 4]
```diff
@@ -19,3 +20,59 @@ test('errors early if access token not provided', () => {
...
+    const client = browserClient({ accessToken });
+
+    request = client.createRequest({
+      method: 'GET',
+      path: '/styles/v1/:ownerId/:styleId',
+      params: { styleId: 'foo' }
+    });
+  });
+
+  test('request.emitter should emit uploadProgress events', () => {
```
Comment (davidtheclark) [inline test/browser-interface.test.js]: This should say "should *not* emit", right?

**Comment by kepta** [comment]
@davidtheclark improved the code, let me know what you think.

**Pull request merged**